### ISSUE TITLE
order of generated code should be deterministic

### ISSUE THREAD

**Issue by GlenDC** [Jumpscale/go-raml#271]
Right now the order of go interface methods, handlerFunc registration calls seems to be random. Perhaps there is even more than that non-deterministic, but that's the only behavior I can observe so far. Which I'm assuming is because we store them in the go-raml generation process in a `map`, which would explain this behavior. I haven't seen the go-raml code, but that is my best guess for now.

This is pretty annoying, as it clutters the history of generated code changes, for no good reason. It also means that one of the blockstor's CI tests fails (which checks for uncommitted changes of generated code).

### GridAPI Example

First Run:

```go
// NodesInterface is interface for /nodes root endpoint
type NodesInterface interface { // ListNodes is the handler for GET /nodes
	// List Nodes
	ListNodes(http.ResponseWriter, *http.Request)
	// GetNode is the handler for GET /nodes/{nodeid}
	// Get detailed information of a node
	GetNode(http.ResponseWriter, *http.Request)
	// ListStoragePools is the handler for GET /nodes/{nodeid}/storagepools
	// List storage pools present in the node
	ListStoragePools(http.ResponseWriter, *http.Request)
	// CreateStoragePool is the handler for POST /nodes/{nodeid}/storagepools
	// Create a new storage pool
	// ... many more interface methods
}
```

Second run:

```go
// NodesInterface is interface for /nodes root endpoint
type NodesInterface interface { // ListNodes is the handler for GET /nodes
	// List Nodes
	ListNodes(http.ResponseWriter, *http.Request)
	// GetNode is the handler for GET /nodes/{nodeid}
	// Get detailed information of a node
	GetNode(http.ResponseWriter, *http.Request)
	// GetCPUInfo is the handler for GET /nodes/{nodeid}/cpus
	// Get detailed information of all CPUs in the node
	GetCPUInfo(http.ResponseWriter, *http.Request)
	// GetNodeOSInfo is the handler for GET /nodes/{nodeid}/info
	// Get detailed information of the os of the node
        // ... many more interface methods
}
```

As you can see, I only show you the first 3 methods of this generated interface, and between 2 runs, the order is already different. The diff between the 2 phases is much bigger than that however, as the order of these methods is non-deterministic at the moment.

### Proposed fix

Right before generation of the values, store the `map` content into a list (of `pairs`), sort that list, and only than do the generation. Based on what the ordering is done is not important, as long as there is a deterministic order.

We applied a similar fix to [go-swagger](https://github.com/go-swagger/go-swagger) a couple of months ago.

**Comment by iwanbk** [comment]
```
Right before generation of the values, store the map content into a list (of pairs), sort that list, and only than do the generation. Based on what the ordering is done is not important, as long as there is a deterministic order.
```
Yep, actually it is already done in many parts of go-raml (check with `grep -En sort -r . | grep Strings`, but looks like we still missed some places.

Can you give your .raml file? Because we already have test fixtures that one of it's purpose is to handle this kind of thing and unfortunately it doesn't catch this issue.

**Comment by GlenDC** [comment]
The raml files can be found here: https://github.com/g8os/blockstor/tree/master/gridapi/gridspec, they are pretty big though. I think it should be fairly simple to create a mini raml file that triggers this issue. Anyway the linked raml files are used as input for the generated code above.

**Comment by iwanbk** [comment]
fixed in #272
